### DISCUSSION TITLE
Stencil Components + Next.js 13 app/ folder

### DISCUSSION THREAD

**Discussion by gabrielleacuba** [vercel/next.js#50306] [App Router]
Can any of you help me? I have my company’s design system that is using the stencil and they work perfectly in the next version that uses the pages/ folder, but trying to use it in the app/ folder in the new 13.4 version the components are loaded but the style is not

If I set the `defineCustomElements` in page layout the component appears and disappears soon after, but in html it stays but its style doesn’t!

![Screen Shot 2023-05-24 at 18 21 26](https://github.com/ionic-team/stencil/assets/43866423/0cf4e8dc-503b-4497-a550-873973bb8e86)

```
app/page.tsx

"use client";
import { MyComponent } from "@my-scope/my-stencil-lib";
import { useEffect } from "react";

export default function Home() {
  useEffect(() => {
    import("@my-scope/my-stencil-lib").then(
      ({ defineCustomElements }) => {
        defineCustomElements();
      }
    );
  }, []);
  return <MyComponent variant="cta">Click me</MyComponent>;
}
```

```
next.config.js 

/** @type {import('next').NextConfig} */
const nextConfig = {
  transpilePackages: ["@my-scope/my-stencil-lib"],
};

module.exports = nextConfig;

```

**Comment by NikolaDonchev** [comment]
I am currently on the same boat.

Have tried to do some debugging yesterday and on next.js 13.4, it cannot render the component and generate the `$vnode$` prop at all that I would usually expect to see when I log the `hostRef`. It gets as far as trying to write to the DOM. Wasn't able to find why it lost the `hostRef` at one point during `setValue`. 

I hope this helps and I'm open to suggestions.

<img width="1633" alt="Screenshot 2023-05-31 at 11 13 20" src="https://github.com/vercel/next.js/assets/3798807/63aaf2a5-74be-4585-8163-c07710e3494b">

**Reply by juanmnoronha** [reply]
Any news in solving this problem? If so, can you share it, please?
